util/kubeconfig/kubeconfig.go

```diff
@@ -54,14 +55,14 @@ func FromSecret(ctx context.Context, c client.Reader, cluster client.ObjectKey)
 }
 
 // New creates a new Kubeconfig using the cluster name and specified endpoint.
-func New(clusterName, endpoint string, caCert *x509.Certificate, caKey crypto.Signer) (*api.Config, error) {
+func New(clusterName, endpoint string, caCert *x509.Certificate, caKey crypto.Signer, keyEncryptionAlgorithm bootstrapv1.EncryptionAlgorithmType) (*api.Config, error) {
```

Comment:
I think let's introduce an options var arg an implement one option to provide a crypto.Signer. 
If the option is not provided let's fallback to the previous behavior

(I want to avoid coupling these funcs to the bootstrap provider and varargs allow a change without affection consumers)

Reply:
I would use this options pattern:
* https://github.com/kubernetes-sigs/cluster-api/blob/d0a809bddb6e76173ad32826bb3f9aa060104664/util/conditions/mirror.go#L32-L52
* https://github.com/kubernetes-sigs/cluster-api/blob/d0a809bddb6e76173ad32826bb3f9aa060104664/util/conditions/options.go#L56
* https://github.com/kubernetes-sigs/cluster-api/blob/d0a809bddb6e76173ad32826bb3f9aa060104664/util/conditions/mirror.go#L60

Reply:
got it thanks.